Composite/Composite/Transforms.fs

```diff
@@ -33,12 +33,60 @@ module Transforms =
             seq {
                     let mutable cur = getNext ()
                     while cur |> Option.isSome do
-                        yield cur |> (function | Some x -> x | None -> failwith "Unexpected situation in enumeration.")
+                        yield cur.Value
                         cur <- getNext ()
-                }
+            }
 
         Array.init numParts (fun _ -> getSeq ())
 
+    let toPaged pageSize inSeq =
+        let rec getPageAndRest numRemains pageAndRest = 
+            match numRemains > 0, pageAndRest with
+            | true, (p, r) -> match r |> Seq.tryHead with
+                                | Some h -> getPageAndRest (numRemains-1) (Array.append p [|h|], r |> Seq.tail)
+                                | None -> pageAndRest
+            | false, pr -> pr
+
+        let rec getPages inSeq2 =
+            seq {
+                    match getPageAndRest pageSize ([||], inSeq2) with
+                    | [||], _ -> yield! []
+                    | p, r -> yield p
+                              yield! getPages r
+            }
+        getPages inSeq
+
+    let toBatched batchSize getElemSize inSeq =
+        // Here we extract a batch from an input sequence.
+        // We do this recursively, passing the amount of remaining
+        // space in batch as a parameter, plus a tuple comprising
+        // the batch under constuction and the remaining sequence
+        // we have to walk through.
+        let rec getBatchAndRest roomLeft batchAndRest = 
+            let (page, rest) = batchAndRest

```

Comment:
The value `page` may be renamed to `batch`.

Reply:
@v-ilin A valid point.